thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -299,6 +302,7 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
     }
     inline const void *nextRow(unsigned outIdx, rowcount_t current)
     {
+        ActivityTimer t(slaveTimerStats, queryTimeActivities());
```

Comment:
Activities downstream from the splitter arm make use of the CSplitterOutput timings. Downstream activities  make use of the queryTotalCycles from the CSplitterOutput (they are not using not NSpitterSlaveActivity::queryTotalCycles).

Reply:
I've created a jira for implementing the blocked time for splitters: https://hpccsystems.atlassian.net/browse/HPCC-32952

Reply:
The activity's slaveTimerStats is used to report the activity's total time and calculate the activity's local time.

Reply:
hm, this doesn't really make sense.
The downstream activities, will correctly use the timings from the CSplitterOutput's, as they are the inputs.

But the splitter activity will use these timings from NSplitterSlaveActivity::prepareInput and NSplitterSlaveActivity::nextRow, but that is not representative of how long this splitter activity took.
NB: NSplitterSlaveActivity::nextRow will not be called at all, if using newSplitter, and even when it is, it will be included in the CSplitterOutput::nextRow time (as will the prepareInput time).

How long did the splitter spend processing rows..?
Is it the sum of the time of it's arms (CSplitterOutput's)?
Maybe, that will include the time from the 1st that started the input (prepareInput), and all the time it spent reading off disk in the spilling case, but without accounting for block time, that will also count a lot of blocked time too, e.g. when arm 1 can't return any rows, because arm 2 is behind and blocking.

I don't think slaveTimerStats as a member in the activity, nor accumulating it there is correct.